### DISCUSSION TITLE
Type-checking in modules with stubs

### DISCUSSION THREAD

**Discussion by asleep-cult** [microsoft/pyright#1878] [Ideas]
When editing a module with stubs, the type checker does not look for type information in said stubs. This means that developers must disable type checking for the file or disable type checking entirely every time they want to switch between editing their stubs and their actual module (this is pretty tedious when you're using vscode). This also means that all the benefits of typings are lost when editing the module. I propose one of the following:
- The type checker could look for type information in both `.py` and `.pyi` files
- The type checker could automatically disable type-checking in `.py` files when the corresponding `.pyi` file is present

**Comment by jakebailey** [comment]
This sounds pretty similar to https://github.com/microsoft/pylance-release/issues/1238.

I do think that there's a case to be made for improving the development experience for stubs next to py files, but what is your rationale for not inlining your types?

**Reply by asleep-cult** [reply]
1. I can use newer typing features without breaking stuff
2. The module and the stubs can be installed separately
3. Type hinting wont interfere with anything at runtime (saves a bit of memory, a bit faster when importing)
4. I'm not fond of the way type hints look
